src/spatialIndex.cc

```diff
@@ -225,6 +226,50 @@ NAN_METHOD(SpatialIndex::queryPoint) {
   info.GetReturnValue().Set(result);
 }
 
+/*
+ * Gets all ids embedding the provided point coordinates
+ * queryIntersect([[lat, lon], [lat, lon], [lat, lon], ...]])
+ *
+ * Returns an array of matching ids as strings
+ */
+NAN_METHOD(SpatialIndex::queryIntersect) {
+  Nan::HandleScope scope;
+  SpatialIndex *spi = Nan::ObjectWrap::Unwrap<SpatialIndex>(info.This());
+
+  // Checks the coordinates parameters validity
+  if (info[0]->IsUndefined() || !info[0]->IsArray()) {
+    info.GetReturnValue().Set(false);
+    return;
+  }
```

Comment:
To be removed: it is never executed thanks to the guards in the JS code. 
If you decide to keep it, then it must return an empty array as a result instead of a boolean.

Reply:
thanks for the feedback. I have updated that check to return an empty array.